oteapi/models/genericconfig.py

```diff
@@ -1,104 +1,125 @@
 """Generic data model for configuration attributes."""
-from typing import TYPE_CHECKING, Iterable, Mapping
+from typing import TYPE_CHECKING, Iterable, Mapping, MutableMapping
 
-from pydantic import BaseModel, Field
-from pydantic.fields import Undefined
+from pydantic import BaseModel, ConfigDict, Field
+from pydantic.fields import PydanticUndefined
 
 if TYPE_CHECKING:  # pragma: no cover
     from typing import Any, Optional, Tuple, Union
 
 
-class AttrDict(BaseModel, Mapping):
+class AttrDict(BaseModel, MutableMapping):
     """An object whose attributes can also be accessed through
     subscription, like with a dictionary."""
 
-    def __contains__(self, name: "Any") -> bool:
-        """Enable using the 'in' operator on this object."""
-        return self.__dict__.__contains__(name)
-
-    def __delitem__(self, key: str) -> None:
-        """Enable deletion access through subscription.
-
-        If the item is a pydantic field, reset it and remove it from the set of set
-        fields. Otherwise, delete the attribute.
-
-        """
-        if key in self.__dict__:
-            if key in self.__fields__:
-                # Reset field to default and remove from set of set fields
-                setattr(self, key, self.__fields__[key].default)
-                self.__fields_set__.remove(key)
-            else:
-                self.__delattr__(key)
-        else:
-            raise KeyError(key)
+    model_config = ConfigDict(
+        extra="allow", validate_assignment=True, arbitrary_types_allowed=True
+    )
 
-    def __getitem__(self, key: str) -> "Any":
-        """Enable read access through subscription."""
-        if key in self.__dict__:
+    # Collection methods
+    def __contains__(self, key: object) -> bool:
+        """Mapping `__contains__`-method."""
```

Comment:
Go over the doc strings to make sure you're referencing the correct class, e.g. here it should be 'Collection'